apache/zeppelin
# [ZEPPELIN-1679] Invalid thrift implementation (developer critical)

### What is this PR for?

Recently, zeppelin added some amazing features. 
https://github.com/apache/zeppelin/pull/1613 
You can easily access spark ui from the zeppelin web.
However, in the implementation of the function, the Thrift interface seems to have arbitrarily modified the automatically generated code.
So if you recreate the Thift interface by calling genthrift.sh.
Zeppelin is not built.
I modified this to get the same result with auto-generated code.


### What type of PR is it?
Bug Fix

### Todos
- [x] add event META_INFOS in RemoteInterpreterService.thrift
- [x] import generate code 

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1679

### How should this be tested?
1. First you need to install thift 0.9.2.
2. Run zeppelin-interpreter / src / main / thrift / generate_thrift.sh.
3. Build full zeppelin.
4. After build, please check the interpreter spark setting in our web ui.
5. Try using the spark ui button.

If everything is running normally, it is a success.
Problems occur in the current master 3.

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no
